DDDEastAnglia.Tests/InMemoryRepository.cs

```diff
@@ -0,0 +1,43 @@
+using System;
+using System.Collections.Concurrent;
+using System.Collections.Generic;
+
+namespace DDDEastAnglia.Tests
+{
+    internal class InMemoryRepository<T>
+    {
+        private readonly Func<T, int> keyFunc;
+        private readonly ConcurrentDictionary<int, T> store = new ConcurrentDictionary<int, T>();
+
+        public InMemoryRepository(Func<T, int> keyFunc)
```

Comment:
It's not immediately obvious to me what I should pass as the `keyFunc`: generics and primitives aren't that expressive. Can that be improved somehow?

On a related note, some guidance on how to write a key function might be good too? I'm guessing `_ => 1` isn't going to be very useful... but equally, I'm not sure what *would* be useful.